circleci/templates/gradle/config_template-v0_1.yml

```diff
@@ -1,159 +1,158 @@
+  Docker Image Build: # Build-Container Image -- Call out what flavor (likely always Docker) and indicate this is the Build of said container/image
+    docker:
+      - image: circleci/ruby:2.4.1
     steps:
-      - run: *env
       - checkout
+      - run: echo "Building Docker Image"
+  Sonar Code Quality Scan: # Quality-Code Scanning -- Call out what flavor (Sonar presently) and indicate Code Quality Scan
+    docker: *image
+    steps:
+    # PR Analysis cannot have branch.target parameter
+    # PR Analysis requires "pullrequest.provider" & "pullrequest.github.reposotiry" to fire decorators back to GitHub
+    # CircleCI limit:  no organization name for easier pointing to github.repository value
       - run:
-          name: Sonar check
+          name: Determining Sonar and Branch Configuration
           command: |
-            if [ $CIRCLE_BRANCH = master ]
-            then
-              if [ $RELEASE = true ]
+              SONAR_PARAMS='';
+              if [[ ! -z $CIRCLE_PULL_REQUEST ]];
               then
-                ./gradlew sonarqube -Dsonar.host.url=https://sonarcloud.io -Dsonar.login=$SONAR_TOKEN -Dsonar.organization=cx -Dsonar.projectKey=multi-int-store
+                SONAR_PARAMS="${SONAR_PARAMS} \
+                -Dsonar.pullrequest.key=${CIRCLE_PULL_REQUEST##*/} \
+                -Dsonar.pullrequest.base=master \
+                -Dsonar.pullrequest.branch=${CIRCLE_BRANCH} \
+                -Dsonar.pullrequest.provider=GitHub
+                -Dsonar.pullrequest.github.repository=connexta/${CIRCLE_PROJECT_REPONAME}"
               fi
-            fi
-  release_tag:
-    docker: *image
-    steps:
-      - run: *env
+              if [[ ${CIRCLE_BRANCH} != 'master' && -z $CIRCLE_PULL_REQUEST ]];
+              then
+                SONAR_PARAMS="${SONAR_PARAMS} \
+                -Dsonar.branch.target=master \
+                -Dsonar.branch.name=${CIRCLE_BRANCH}"
+              fi
+              echo "export SONAR_PARAMS='${SONAR_PARAMS}'" >> $BASH_ENV
+              source $BASH_ENV
       - checkout
+        # sonar.pullrequest.base should be your default branch but may not be "master"
+        # sonar.login is the API token we can use and here is configured for a CircleCI Env Variable
+        # sonar.organization "cx" is the Connexta Organization
+        # CircleCI Env. Variable CIRCLE_PULL_REQUEST only works for "forked" repos
       - run:
-          name: Release Tag
+          name: Scanning
           command: |
-            if [ $RELEASE = true ]
-            then
-              git checkout $RELEASE_COMMIT
-              git remote add ssh-origin git@github.com:connexta/$CIRCLE_PROJECT_REPONAME.git
-              echo ssh -i $GITHUB_KEY -l git -o StrictHostKeyChecking=no \\"\\$@\\" > run_ssh.sh
-              chmod +x run_ssh.sh
-              git push ssh-origin HEAD:$CIRCLE_BRANCH && git push ssh-origin $RELEASE_TAG
-            fi
-  deploy:
-    docker: *image
+            echo $SONAR_PARAMS;
```

Comment:
While this structure works for maven, I'm not certain it works for gradle.  Have you tested this?
We may need to add additional logic to allow sonar to work properly with Gradle that isn't a direct port over.

Reply:
Do we know if the command line arguments also still work with this or do they get ignored?